syntax = "proto3";

package svg_proto;

message Rgb {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
}

message Rgba {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
    double opacity = 4;
}

message Point {
    double x = 1;
    double y = 2;
}

message Color {
    oneof variant_color {
        Rgb color_rgb = 1;
        Rgba color_rgba = 2;
        string color_string = 3;
    }
}